Library/Homebrew/caveats.rb

```diff
@@ -82,45 +80,38 @@ def bash_completion_caveats
     EOS
   end
 
-  def zsh_completion_caveats
+  def zsh_caveats
     return unless keg
-    return unless keg.completion_installed?(:zsh)
 
-    <<-EOS.undent
-      zsh completion has been installed to:
-        #{HOMEBREW_PREFIX}/share/zsh/site-functions
-    EOS
-  end
-
-  def fish_completion_caveats
-    return unless keg
-    return unless keg.completion_installed?(:fish)
-    return unless which("fish")
-
-    <<-EOS.undent
-      fish completion has been installed to:
-        #{HOMEBREW_PREFIX}/share/fish/vendor_completions.d
-    EOS
-  end
+    comp = keg.completion_installed?(:zsh)
+    func = keg.zsh_functions_installed?
+    return unless comp || func
 
-  def zsh_function_caveats
-    return unless keg
-    return unless keg.zsh_functions_installed?
+    installed = []
+    installed << "completions" if comp
+    installed << "functions" if func
```

Comment:
I thought about it, but we'd have to do some messy things like `keg.send("#{shell}_functions_installed?")` and the directory paths are different:
```
share/fish/vendor_completions.d
```
versus
```
share/zsh/site-functions